src/streams.cpp

```diff
@@ -29,18 +35,17 @@ void AutoFile::seek(int64_t offset, int origin)
     if (std::fseek(m_file, offset, origin) != 0) {
         throw std::ios_base::failure(feof() ? "AutoFile::seek: end of file" : "AutoFile::seek: fseek failed");
     }
-}
-
-int64_t AutoFile::tell()
-{
-    if (IsNull()) {
-        throw std::ios_base::failure("AutoFile::tell: file handle is nullptr");
-    }
-    int64_t r{std::ftell(m_file)};
-    if (r < 0) {
-        throw std::ios_base::failure("AutoFile::tell: ftell failed");
+    if (origin == SEEK_SET) {
+        m_position = offset;
+    } else if (origin == SEEK_CUR) {
+        m_position += offset;
+    } else {
+        int64_t r{std::ftell(m_file)};
+        if (r < 0) {
+            throw std::ios_base::failure("AutoFile::seek: ftell failed");
+        }
+        m_position = r;
```

Comment:
Only commenting because this confused me:

This makes sense since the third possibility here is that `origin == SEEK_END`, in which case the only way for `fseek` to have returned zero is if `offset` is a negative number offset from the end of the file, and `m_position = length_of_file + negative_offset`

But, in order to get the length of the fil you need to `fseek(m_file, 0, SEEK_END)` and `ftell(m_file)`.

So, just `ftell` here is the best way to get the offset.